File: app/src/main/java/com/example/villageera/SplashScreenActivity.java
package com.example.villageera;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;

public class SplashScreenActivity extends AppCompatActivity {
    //Declaring variables
    private static int SPLASH_TIMER = 2500;
    Animation bottom_anim, side_anim;
    ImageView Image;
    TextView logoOne, logoTwo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);
        //hiding the action bar from the activity
        if (getSupportActionBar() != null) {
            getSupportActionBar().hide();
        }

        //hooks
        Image = findViewById(R.id.main_img);
        logoOne = findViewById(R.id.logo_name_fir);
        logoTwo = findViewById(R.id.logo_name_sec);

        //Animation hooks
        side_anim = AnimationUtils.loadAnimation(this, R.anim.side_anim);
        bottom_anim = AnimationUtils.loadAnimation(this, R.anim.bottom_anim);

        //set animations to elements
        Image.setAnimation(side_anim);
        logoOne.setAnimation(bottom_anim);
        logoTwo.setAnimation(bottom_anim);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                startActivity(new Intent(SplashScreenActivity.this, DashBoardActivity.class));
                finish();
            }
        }, SPLASH_TIMER);
    }
}

File: app/src/main/java/com/example/villageera/SplashScreenActivity.kt
package com.example.villageera

import android.content.Intent
import android.os.Bundle
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.example.villageera.databinding.ActivitySplashScreenBinding
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class SplashScreenActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySplashScreenBinding

    private val splashTime = 2500L
    private lateinit var bottomAnim: Animation
    private lateinit var sideAnim: Animation

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivitySplashScreenBinding.inflate(layoutInflater)
        setContentView(binding.root)

        supportActionBar?.hide()

        // Load animations
        sideAnim = AnimationUtils.loadAnimation(this, R.anim.side_anim)
        bottomAnim = AnimationUtils.loadAnimation(this, R.anim.bottom_anim)

        // Apply animations
        binding.mainImg.startAnimation(sideAnim)
        binding.logoNameFir.startAnimation(bottomAnim)
        binding.logoNameSec.startAnimation(bottomAnim)

        // Delay using coroutines (modern replacement for Handler)
        lifecycleScope.launch {
            delay(splashTime)
            startActivity(Intent(this@SplashScreenActivity, DashBoardActivity::class.java))
            finish()
        }
    }
}
